js/examples/qa_agent/demo.mjs

```diff
@@ -0,0 +1,67 @@
+import { groq } from "@ai-sdk/groq";
+import { VercelAIToolSet } from "composio-core";
+import dotenv from "dotenv";
+import { generateText } from "ai";
+dotenv.config();
+
+// Setup toolset
+const toolset = new VercelAIToolSet({
+  apiKey: process.env.COMPOSIO_API_KEY,
+});
+
+const appName = "codeinterpreter";
+
+async function setupUserConnectionIfNotExists(entityId) {
+  const entity = await toolset.client.getEntity(entityId);
+  const connection = await entity.getConnection(appName);
+
+  if (!connection) {
+    // Initiate a new connection if it doesn't exist
+    const newConnection = await entity.initiateConnection(appName);
+    console.log("Log in via: ", newConnection.redirectUrl);
+    return newConnection.waitUntilActive(60);
+  }
+
+  return connection;
+}
+
+async function executeAgent(entityName) {
+  // Setup entity and ensure connection
+  const entity = await toolset.client.getEntity(entityName);
+  await setupUserConnectionIfNotExists(entity.id);
+
+  // Retrieve tools for the specified app
+  const tools = await toolset.getTools({ apps: ['CODEANALYSISTOOL',appName] }, entity.id);
+  const task = "Your job is to be a QA test engineer.";
+  const code = `
+class Calculator:
```

Comment:
The Calculator class should include input validation to ensure arguments are numeric. Consider adding a helper method:

```python
def _validate_numbers(self, *args):
    """Validate that all arguments are numbers.
    
    Args:
        *args: Variable number of arguments to validate
        
    Raises:
        TypeError: If any argument is not a number
    """
    for arg in args:
        if not isinstance(arg, (int, float)):
            raise TypeError(f"Expected number but got {type(arg)}")
```

And use it in each operation method:
```python
def add(self, a, b):
    self._validate_numbers(a, b)
    return a + b
```